modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js

```diff
@@ -67,30 +67,19 @@ class ContributorBuilder extends React.Component {
 	}
 
 	_onCriteriaChange = (criteriaChange, index) => {
-		this.setState(
-			state => {
-				let newState;
-
-				if (state.editing !== index) {
-					newState = state;
-				}
-				else {
-					newState = {
-						contributors: state.contributors.map(
-							(contributor, i) => {
-								return (index === i) ? {
-									...contributor,
-									criteriaMap: criteriaChange,
-									query: buildQueryString([criteriaChange])
-								} : contributor;
-							}
-						)
-					};
-				}
-
-				return newState;
-			}
-		);
+		if (this.state.editing === index) {
```

Comment:
I ran into really weird problems in the past relying on `this.state` values at `this.setState`, same with props. It is recommended to pass a function.
More info here: https://reactjs.org/docs/faq-state.html#why-is-setstate-giving-me-the-wrong-value